File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/javascript/find_shortest_string.js
function findShortestString(arr) {
  // type your code here
  let hh=[]
 for(let str of arr){
   let ans=str.split('')

    hh.push(ans.length)
 }
 let aaa=Math.min(...hh)
 let bbb=hh.indexOf(aaa)
  return arr[bbb]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'a'");
  console.log("=>", findShortestString(['aaa', 'a', 'bb', 'ccc']));

  console.log("");

  console.log("Expecting: 'hi'");
  console.log("=>", findShortestString(['cat', 'hi', 'dog', 'an']));

  console.log("");

  console.log("Expecting: 'lily'");
  console.log("=>", findShortestString(['flower', 'juniper', 'lily', 'dadelion']));

  // BENCHMARK HERE
}

module.exports = findShortestString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/ruby/find_shortest_string.rb
def find_shortest_string(arr)
  # type your code in here
  newA= arr[0]
  arr.each do |string|
    if string.size < newA.length
      newA=string
    end
    newA
  end
  newA
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'a'"
  puts "=>", find_shortest_string(['aaa', 'a', 'bb', 'ccc'])

  puts

  puts "Expecting: 'hi'"
  puts "=>", find_shortest_string(['cat', 'hi', 'dog', 'an'])

  puts

  puts "Expecting: 'lily'"
  puts "=>", find_shortest_string(['flower', 'juniper', 'lily', 'dadelion'])

  # Don't forget to add your own!

  puts "Expecting: 'fl'"
  puts "=>", find_shortest_string(['fl', 'juniper', 'lily', 'dadelion'])

  # BENCHMARK HERE
end

# Please add your pseudocode to this file
# And a written explanation of your solution
